### ISSUE TITLE
`yarn dev` produce error (`Cannot find module 'typescript'`) after create project and install

### ISSUE THREAD

**Issue by ayanamists** [slidevjs/slidev#1223]
**Describe the bug**
After run `yarn create slidev` and `yarn`, `yarn dev` or `yarn build` will produce such error:

```
❯ yarn build
yarn run v1.22.19
$ slidev build


  ●■▲
  Slidev  v0.46.1 

  theme       @slidev/theme-seriph
  css engine  unocss
  entry       /home/ayanamists/repo/sicp/lab10/slides.md
vite v5.0.10 building for production...
✓ 37 modules transformed.
[unplugin-vue-markdown] Cannot find module 'typescript'
Require stack:
- /home/ayanamists/repo/sicp/lab10/node_modules/@typescript/twoslash/dist/twoslash.cjs.development.js
```

**To Reproduce**
1. Create a project by `yarn create slidev`
2. Run `yarn` to install the dependencies
3. Run `yarn dev` or `yarn build`

**Desktop (please complete the following information):**
 - OS: Ubuntu 22.04.3, Kernel: 6.2.0-39-generic
 - Yarn version: 1.22.19 
 - Slidev version: 0.46.1
 
 **Full Stack Trace:**
```
❯ yarn build
yarn run v1.22.19
$ slidev build


  ●■▲
  Slidev  v0.46.1 

  theme       @slidev/theme-seriph
  css engine  unocss
  entry       /home/ayanamists/repo/sicp/lab10/slides.md
vite v5.0.10 building for production...
✓ 37 modules transformed.
[unplugin-vue-markdown] Cannot find module 'typescript'
Require stack:
- /home/ayanamists/repo/sicp/lab10/node_modules/@typescript/twoslash/dist/twoslash.cjs.development.js
file: /@slidev/slides/5.md
Error: Cannot find module 'typescript'
Require stack:
- /home/ayanamists/repo/sicp/lab10/node_modules/@typescript/twoslash/dist/twoslash.cjs.development.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)
    at Module._load (node:internal/modules/cjs/loader:922:27)
    at Module.require (node:internal/modules/cjs/loader:1143:19)
    at require (node:internal/modules/cjs/helpers:119:18)
    at twoslasher (/home/ayanamists/repo/sicp/lab10/node_modules/@typescript/twoslash/dist/twoslash.cjs.development.js:606:81)
    at Object.preprocess (file:///home/ayanamists/repo/sicp/lab10/node_modules/shikiji-twoslash/dist/core.mjs:779:28)
    at codeToHtml (file:///home/ayanamists/repo/sicp/lab10/node_modules/shikiji-core/dist/index.mjs:4590:42)
    at Object.codeToHtml (file:///home/ayanamists/repo/sicp/lab10/node_modules/shikiji-core/dist/index.mjs:4912:40)
    at markdownit.options.highlight (file:///home/ayanamists/repo/sicp/lab10/node_modules/markdown-it-shikiji/dist/core.mjs:56:24)
    at default_rules.fence (/home/ayanamists/repo/sicp/lab10/node_modules/markdown-it/lib/renderer.js:53:27)
    at Renderer.render (/home/ayanamists/repo/sicp/lab10/node_modules/markdown-it/lib/renderer.js:332:28)
    at MarkdownIt.render (/home/ayanamists/repo/sicp/lab10/node_modules/markdown-it/lib/index.js:544:24)
    at file:///home/ayanamists/repo/sicp/lab10/node_modules/unplugin-vue-markdown/dist/chunk-EES4WMNF.js:242:25
    at Object.transform (file:///home/ayanamists/repo/sicp/lab10/node_modules/unplugin-vue-markdown/dist/chunk-EES4WMNF.js:413:37) {
  code: 'PLUGIN_ERROR',
  requireStack: [
    '/home/ayanamists/repo/sicp/lab10/node_modules/@typescript/twoslash/dist/twoslash.cjs.development.js'
  ],
  id: '/@slidev/slides/5.md',
  hook: 'transform',
  pluginCode: 'MODULE_NOT_FOUND',
  plugin: 'unplugin-vue-markdown',
  watchFiles: [
    '/home/ayanamists/repo/sicp/lab10/index.html',
    '/home/ayanamists/repo/sicp/lab10/node_modules/@slidev/client/main.ts',
    '/home/ayanamists/repo/sicp/lab10/node_modules/vue/dist/vue.runtime.esm-bundler.js',
    '/home/ayanamists/repo/sicp/lab10/node_modules/@unhead/vue/dist/index.mjs',
    '/home/ayanamists/repo/sicp/lab10/node_modules/@slidev/client/setup/main.ts',
    '/home/ayanamists/repo/sicp/lab10/node_modules/@slidev/client/App.vue',
    '/home/ayanamists/repo/sicp/lab10/node_modules/@slidev/client/routes.ts',
    '/home/ayanamists/repo/sicp/lab10/node_modules/@slidev/client/modules/directives.ts',
    '/home/ayanamists/repo/sicp/lab10/node_modules/@slidev/client/modules/context.ts',
    '/home/ayanamists/repo/sicp/lab10/node_modules/@unocss/reset/tailwind.css',
    '/home/ayanamists/repo/sicp/lab10/node_modules/@slidev/client/styles/vars.css',
    '/home/ayanamists/repo/sicp/lab10/node_modules/@slidev/client/styles/index.css',
    '/home/ayanamists/repo/sicp/lab10/node_modules/@slidev/client/styles/code.css',
    '/home/ayanamists/repo/sicp/lab10/node_modules/@slidev/client/styles/katex.css',
    '/home/ayanamists/repo/sicp/lab10/node_modules/@slidev/client/styles/transitions.css',
    '/home/ayanamists/repo/sicp/lab10/node_modules/@slidev/theme-seriph/styles/index.ts',
    '/home/ayanamists/repo/sicp/lab10/node_modules/katex/dist/katex.min.css',
    '/home/ayanamists/repo/sicp/lab10/node_modules/shikiji-twoslash/style-rich.css',
    '/home/ayanamists/repo/sicp/lab10/node_modules/unhead/dist/index.mjs',
    '/home/ayanamists/repo/sicp/lab10/node_modules/@unhead/vue/dist/shared/vue.cf295fb1.mjs',
    '/home/ayanamists/repo/sicp/lab10/node_modules/@unhead/vue/dist/shared/vue.f36acd1f.mjs',
    '/home/ayanamists/repo/sicp/lab10/node_modules/@unhead/shared/dist/index.mjs',
    '/home/ayanamists/repo/sicp/lab10/node_modules/@vue/runtime-dom/dist/runtime-dom.esm-bundler.js',
    '/home/ayanamists/repo/sicp/lab10/node_modules/@vueuse/motion/dist/index.mjs',
    '/home/ayanamists/repo/sicp/lab10/node_modules/vue-starport/dist/index.mjs',
    '/home/ayanamists/repo/sicp/lab10/node_modules/@antfu/utils/dist/index.mjs',
    '/home/ayanamists/repo/sicp/lab10/node_modules/@slidev/client/logic/nav.ts',
    '/home/ayanamists/repo/sicp/lab10/node_modules/@slidev/client/constants.ts',
    '/home/ayanamists/repo/sicp/lab10/node_modules/@slidev/client/internals/Play.vue',
    '/home/ayanamists/repo/sicp/lab10/node_modules/vue-router/dist/vue-router.mjs',
    '/home/ayanamists/repo/sicp/lab10/node_modules/@slidev/client/internals/PresenterPrint.vue',
    '/home/ayanamists/repo/sicp/lab10/node_modules/@slidev/client/internals/Print.vue',
    '/home/ayanamists/repo/sicp/lab10/node_modules/@slidev/client/internals/Presenter.vue',
    '/home/ayanamists/repo/sicp/lab10/node_modules/@slidev/client/logic/dark.ts',
    '/home/ayanamists/repo/sicp/lab10/node_modules/@slidev/client/composables/useContext.ts',
    '/home/ayanamists/repo/sicp/lab10/node_modules/@slidev/client/setup/root.ts'
  ]
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by ligdy7** [comment]
I have the same problem

To Reproduce
Steps to reproduce the behavior:

Create a project by yarn create slidev
Run yarn to install the dependencies
Run yarn dev or yarn build
Desktop (please complete the following information):

system：windows
Yarn version: 1.22.21
Slidev version: 0.46.1

```
PS E:\github\share-nestjs\nest-share> yarn dev
yarn run v1.22.21
$ slidev --open


  ●■▲
  Slidev  v0.46.1

  theme       @slidev/theme-seriph
  css engine  unocss
  entry       E:\github\share-nestjs\nest-share\slides.md

  public slide show   > http://localhost:3030/
  presenter mode      > http://localhost:3030/presenter/
  remote control      > pass --remote to enable

  shortcuts           > restart | open | edit | quit
14:34:26 [vite] Pre-transform error: Cannot find module 'typescript'
Require stack:
- E:\github\share-nestjs\nest-share\node_modules\@typescript\twoslash\dist\twoslash.cjs.development.js
14:34:29 [vite] Pre-transform error: Cannot find module 'typescript'
Require stack:
- E:\github\share-nestjs\nest-share\node_modules\@typescript\twoslash\dist\twoslash.cjs.development.js (x2)
14:34:29 [vite] Pre-transform error: Cannot find module 'typescript'
Require stack:
- E:\github\share-nestjs\nest-share\node_modules\@typescript\twoslash\dist\twoslash.cjs.development.js (x3)
14:34:31 [vite] Pre-transform error: Cannot find module 'typescript'
Require stack:
- E:\github\share-nestjs\nest-share\node_modules\@typescript\twoslash\dist\twoslash.cjs.development.js (x6)
14:34:31 [vite] Pre-transform error: Cannot find module 'typescript'
Require stack:
- E:\github\share-nestjs\nest-share\node_modules\@typescript\twoslash\dist\twoslash.cjs.development.js (x7)
14:34:31 [vite] Pre-transform error: Cannot find module 'typescript'
Require stack:
- E:\github\share-nestjs\nest-share\node_modules\@typescript\twoslash\dist\twoslash.cjs.development.js (x8)
14:34:31 [vite] Internal server error: Cannot find module 'typescript'
Require stack:
- E:\github\share-nestjs\nest-share\node_modules\@typescript\twoslash\dist\twoslash.cjs.development.js
  Plugin: unplugin-vue-markdown
  File: /@slidev/slides/5.md
      at Module._resolveFilename (node:internal/modules/cjs/loader:1134:15)
      at Module._load (node:internal/modules/cjs/loader:975:27)
```

**Comment by ayanamists** [comment]
> I have the same problem
> 
> To Reproduce Steps to reproduce the behavior:
> 
> Create a project by yarn create slidev Run yarn to install the dependencies Run yarn dev or yarn build Desktop (please complete the following information):
> 
> system：windows Yarn version: 1.22.21 Slidev version: 0.46.1
> 
> ```
> PS E:\github\share-nestjs\nest-share> yarn dev
> yarn run v1.22.21
> $ slidev --open
> 
> 
>   ●■▲
>   Slidev  v0.46.1
> 
>   theme       @slidev/theme-seriph
>   css engine  unocss
>   entry       E:\github\share-nestjs\nest-share\slides.md
> 
>   public slide show   > http://localhost:3030/
>   presenter mode      > http://localhost:3030/presenter/
>   remote control      > pass --remote to enable
> 
>   shortcuts           > restart | open | edit | quit
> 14:34:26 [vite] Pre-transform error: Cannot find module 'typescript'
> Require stack:
> - E:\github\share-nestjs\nest-share\node_modules\@typescript\twoslash\dist\twoslash.cjs.development.js
> 14:34:29 [vite] Pre-transform error: Cannot find module 'typescript'
> Require stack:
> - E:\github\share-nestjs\nest-share\node_modules\@typescript\twoslash\dist\twoslash.cjs.development.js (x2)
> 14:34:29 [vite] Pre-transform error: Cannot find module 'typescript'
> Require stack:
> - E:\github\share-nestjs\nest-share\node_modules\@typescript\twoslash\dist\twoslash.cjs.development.js (x3)
> 14:34:31 [vite] Pre-transform error: Cannot find module 'typescript'
> Require stack:
> - E:\github\share-nestjs\nest-share\node_modules\@typescript\twoslash\dist\twoslash.cjs.development.js (x6)
> 14:34:31 [vite] Pre-transform error: Cannot find module 'typescript'
> Require stack:
> - E:\github\share-nestjs\nest-share\node_modules\@typescript\twoslash\dist\twoslash.cjs.development.js (x7)
> 14:34:31 [vite] Pre-transform error: Cannot find module 'typescript'
> Require stack:
> - E:\github\share-nestjs\nest-share\node_modules\@typescript\twoslash\dist\twoslash.cjs.development.js (x8)
> 14:34:31 [vite] Internal server error: Cannot find module 'typescript'
> Require stack:
> - E:\github\share-nestjs\nest-share\node_modules\@typescript\twoslash\dist\twoslash.cjs.development.js
>   Plugin: unplugin-vue-markdown
>   File: /@slidev/slides/5.md
>       at Module._resolveFilename (node:internal/modules/cjs/loader:1134:15)
>       at Module._load (node:internal/modules/cjs/loader:975:27)
> ```
> 
> **But when I use npm to build the project, it is fine**

It seems that this problem is caused by vite with yarn.

**Comment by antfu** [comment]
Hopefully this should be fixed in 0.47
